dev/validate_bom__config.py

```diff
@@ -257,6 +298,34 @@ def add_config(cls, options, script):
     script.append(hal)
 
 
+class ArtifactConfigurator(Configurator):
+  """Controls hal config artifact for Spinnaker artifact ."""
+
+  HELPERS = {
+    'gcs': GcsArtifactStorageConfiguratorHelper,
+  }
+
+  def init_argument_parser(self, parser):
+    """Implements interface."""
+    for helper in self.HELPERS.values():
+      helper.init_argument_parser(parser)
+
+  def validate_options(self, options):
```

Comment:
HELPERS is a map but you are treating it as a list.
I presume it is a map for the same reason the storage service is a map.
However even so you are not using it that way. Other than init, you'll never iterate over all the values. It only works now because there is one element in the map -- adding another will break all the code here.

I'd propose either adding the configuration option, of which there is only one choice (and having the remaining code read out the option) or using a class constant as a place holder for that option, whose value is 'gcs' and using that symbol for the HELPERS lookup key. Otherwise get rid of the map and replace it with a variable or use the raw value and refactor later.

Reply:
Ah I see.  I believe the intention is that multiple artifact configurators could all be enabled at once.  A test might need artifacts from multiple storage providers for a single scenario.  With that in mind I've converted the HELPERS map to a list and removed the `raise ValueError` from `GcsArtifactStorageConfiguratorHelper.validate_options()`, which should prevent an explosion from happening during configuration.